R/package-deps.r

```diff
@@ -44,6 +44,44 @@ parse_deps <- function(string) {
   deps[names != "R", ]
 }
 
+# Takes a dependency data frame generated by the `desc` package.
+deps_check_installed <- function(path, deps, call = caller_env()) {
+  if (!nrow(deps)) {
+    return()
+  }
+
+  pkg <- deps$package
+  ver <- deps$version
+
+  # Recreate `pkg (>= ver)` strings
+  has_version <- ver != "*"
+  pkg[has_version] <- sprintf(
+    "%s (%s)",
+    pkg[has_version],
+    ver[has_version]
+  )
+
+  # Outdated and missing dependencies are installed using pak if
+  # installed. If not, the remotes package is used if installed.
+  # Otherwise `install.packages()` is used as a last resort but this
+  # method does not support Remotes fields.
+  action <- function(pkg, ...) {
+    if (is_installed("pak")) {
+      deps <- pak::pkg_deps(path, upgrade = FALSE)
+      deps <- deps[deps$package %in% pkg, ]
+      pak::pak(deps$ref, ask = FALSE)
```

Comment:
This does two rounds of dependency lookups, which is not ideal.

Reply:
Do you have a fix to suggest? This is getting the refs of the package names in `pkg` which are potentially Remotes refs, and then installing them. Is there a way to do these two steps in one go?

Reply:
Why not just call 
```r
pkg_install("deps::.")
```

Reply:
To avoid having to reinstall dependencies from github when pak does not recognise that a local or dev install is sufficient.